React Native 0.68 Update

ksiler opened the pull request:
### Platforms Impacted
- [x] iOS
- [x] macOS
- [x] win32 (Office)
- [x] windows
- [x] android

Waiting to merge until Office bumps to 0.68.

### Description of changes

- Bump react-native from 0.66 to 0.68
- Add rnx-kit config sections for all packages to help with dependency management (see https://microsoft.github.io/rnx-kit/docs/guides/dependency-management)
  - For reference, almost all packages I added an rnx-config section to have this:
   ``` 
  "rnx-kit": {
    "reactNativeVersion": "^0.68",
    "reactNativeDevVersion": "^0.68",
    "kitType": "library",
    "capabilities": [
      "core",
      "core-android",
      "core-ios",
      "react"
    ]
  }
- Ran `yarn rnx-dep-check --write` at root to fix dependencies for RN 0.68

### Verification

- iOS: Not yet tested *
- macOS: Not yet tested *
- win32 (Office): Need to re-test for new changes
- windows: Need to re-test for new changes
- android: Not yet tested *

*hoping someone could help me out with testing these platforms

### Pull request checklist

This PR has considered (when applicable):
- [x] Automated Tests
- [ ] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

rurikoaraki commented:
We should hold off on this until Office bumps to 0.68.

rurikoaraki commented on packages/utils/tokens/package.json:
```diff
@@ -32,13 +32,25 @@
-    "@types/react-native": "^0.66.0",
     "@fluentui-react-native/scripts": "^0.1.1",
-    "react-native": "^0.66.0"
+    "@types/react-native": "^0.66.0",
```
The types package should be bumped to 0.68 as well. The numbers of the types package generally match the RN version number they support.

ksiler replied:
Thanks for the callout! I'll go update that.

Saadnajmi replied:
@kelset is this something rnx-dep-check should have done but missed?

rurikoaraki commented on packages/utils/tokens/package.json:
```diff
@@ -32,13 +32,25 @@
   "author": "",
-  "license": "MIT"
+  "license": "MIT",
+  "rnx-kit": {
```
Curious if we need the rnx-kit field on every package? Not really sure what this does.

Saadnajmi replied:
This makes it so that when you do `npx rnx-dep-check --write`, it updates the dependencies on _every_ package with an rnx-config. Best case, this would mean that a RN version upgrade can be done by that single command.

rurikoaraki commented on apps/fluent-tester/package.json:
```diff
@@ -101,32 +102,22 @@
     "flow-bin": "^0.113.0",
     "metro-config": "^0.66.2",
```
I think metro needs to be bumped to 0.67 per https://reactnative.dev/blog/2022/03/30/version-068

Saadnajmi replied:
Hmm, rnxx-kit bug?

ksiler replied:
Not a bug. This gets resolved to the correct version when adding "metro-config" and "metro-react-native-babel-preset" to the "capabilities" section in the rnx-kit section below. Just didn't realize I needed to add them :)

ksiler replied:
Actually, it seems that it won't automatically update "metro-react-native-babel-preset". Not sure why but I'll manually update it.

Saadnajmi commented on apps/fluent-tester/package.json:
```diff
@@ -101,32 +102,22 @@
-    "react-native-macos": "^0.66.0",
-    "react-native-svg": "^12.3.0",
-    "react-native-windows": "^0.66.0"
+    "@office-iss/react-native-win32": "^0.66.0"
```
Should be 0.68?

ksiler replied:
Thanks, updated this.

Saadnajmi commented on packages/components/Avatar/package.json:
```diff
@@ -36,18 +36,31 @@
     "@fluentui-react-native/theme-tokens": "^0.19.0",
     "@fluentui-react-native/tokens": "^0.16.0",
-    "@fluentui-react-native/use-styling": "^0.9.0",
-    "react-native-svg": "^12.3.0"
```
Interesting this moved to devDependencies?

ksiler replied:
Yeah, I'm not sure why. It's clearly used as part of the Avatar component. I can add it back.

Saadnajmi commented on packages/theming/theming-utils/package.json:
```diff
@@ -30,16 +30,29 @@
     "@fluentui-react-native/scripts": "^0.1.1",
-    "react-native": "^0.66.0",
-    "react-native-windows": "^0.66.0"
+    "@types/react-native": "^0.66.0",
```
Should also be 0.68?

ksiler replied:
Yeah, I'm in the process of updating it across the repo. Seems that rnx-kit did not catch it.

ksiler replied:
Types are updated to 0.68. I'll need to check if rnx-kit has a capability to update that. Still learning how it works.

rurikoaraki commented:
I would have expected to see a few changes in our root package.json as well, i.e. metro-react-native-babel-preset

rurikoaraki commented:
And our script folder's package.json should update their version of @react-native-community/cli

Seems like the update command skipped private packages?

ksiler commented:
> I would have expected to see a few changes in our root package.json as well, i.e. metro-react-native-babel-preset

Thanks, I updated this. Seems that rnx-kit doesn't update this automatically but it will for `metro-config` and `metro-react-native-babel-transformer`

rurikoaraki commented:
> > I would have expected to see a few changes in our root package.json as well, i.e. metro-react-native-babel-preset
> 
> Thanks, I updated this. Seems that rnx-kit doesn't update this automatically but it will for `metro-config` and `metro-react-native-babel-transformer`

Looks like this is the capability name for that package: babel-preset-react-native

List of all capabilities:
https://github.com/microsoft/rnx-kit/tree/main/packages/dep-check#capabilities

ksiler commented:
> And our script folder's package.json should update their version of @react-native-community/cli
> 
> ~Seems like the update command skipped private packages?~ actually this isn't true, i guess just certain dependencies?

Yeah, seems that rnx-kit will update packages based on profiles they define: https://github.com/microsoft/rnx-kit/blob/main/packages/dep-check/src/presets/microsoft/profile-0.68.ts

@react-native-community/cli doesn't seem to be included so I manually updated it.

Saadnajmi commented:
I'm quite impressed all 3 platforms with E2E tests are passing CI right now....

Saadnajmi commented:
Was waiting for builds so I tested this PR. iOS and macOS indeed do compile  and run fine locally. iOS also now runs codeine so there's some new-architecture sneaking it's way in too :)

Saadnajmi reviewed (approved):
Note that we're at 0.68 internally, can we merge this? :)

ksiler commented:
> Note that we're at 0.68 internally, can we merge this? :)

I'm currently merging latest from main and working through some errors.

rurikoaraki commented:
> > Note that we're at 0.68 internally, can we merge this? :)
> 
> I'm currently merging latest from main and working through some errors.

We should make sure the assert on boot is fixed for the latest version of rex.

Saadnajmi commented:
It also might be worth waiting a week or two to make sure 0.68 "sticks" downstream, now that I think about it. Thanks for the updates!

ksiler commented:
Just merged main and added a couple more updates:

- Added rnx-kit configs to @fluentui-react-native/experimental-radio-group and @fluentui-react-native/dropdown
- Added "@types/react": "^17.0.2" to the list of resolutions in package.json to ensure we're relying on the same major version of @types/react
  - @types/enzyme and @types/react-test-renderer were bringing in version 18.0.19 which were causing ts issues
  - `error TS2786: 'View' cannot be used as a JSX component.` in useSlot.test.tsx
- Changed @office-iss/rex-win32 from "0.66.17-devmain.15229.10000" to "0.68.15-devmain.15710.10000"
  - asserts on boot not fixed yet, so we'll probably need to wait for the fix (or else our E2E tests will fail)
- Changed react-native-svg from "^12.3.0" to "12.3.0" due to a type error ("^12.3.0" was resolving to "12.4.4" which uses `ColorValue` for the `Stop` prop which isn't supported natively on win32 - see #866)
  - `ERR! src/Shimmer.win32.tsx:57:28 - error TS2769: No overload matches this call.
ERR!   Overload 1 of 2, '(props: StopProps | Readonly<StopProps>): Stop', gave the following error.
ERR!     Type 'string | number' is not assignable to type 'ColorValue'.
ERR!   Overload 2 of 2, '(props: StopProps, context: any): Stop', gave the following error.
ERR!     Type 'string | number' is not assignable to type 'ColorValue'.`
- Ran `yarn rnx-dep-check --write`

ksiler commented on apps/fluent-tester/package.json:
```diff
@@ -92,45 +93,35 @@
-    "react-native": "^0.66.0",
-    "react-native-macos": "^0.66.0",
-    "react-native-svg": "^12.3.0",
-    "react-native-windows": "^0.66.0"
```
Just double checking - does it make sense to remove all these dependencies from `peerDependencies` in the fluent-tester package?

rurikoaraki replied:
If they're all being moved to dependencies then yeah, that should be fine.

ksiler commented on packages/components/Callout/package.json:
```diff
@@ -39,12 +39,24 @@
-    "react": ">=17.0.2",
-    "react-native": ">=0.66.0"
+    "react": "17.0.2",
+    "react-native": "^0.68.0"
```
For "react" should we use ">=17.0.2" or "17.0.2"? And for "react-native" should it be ">=0.68.0" or "^0.68.0"?

This question applies to multiple packages. `yarn rnx-dep-check` changed it to what you see above.

Saadnajmi replied:
I think it should be "17.0.2", React Native seems to want a specific version of React down to the patch. 
I've seen "^" more than ">=", and I think our use of the latter may have caused some bugs with publishing at times?
In general I trust the output of `rnx-dep-check` more than a convention we've had in our repo, since it's used across multiple repos to manage dependencies

ksiler replied:
Alright, thanks. I'll keep it as-is then.

ksiler commented:
@rurikoaraki @Saadnajmi I'm getting pretty close to passing all checks but the Windows PR keeps failing. When I run `yarn windows --logging` locally, I get this error:

```
 × Building Solution
 × Build failed with message 5:10>C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2302,5): error MSB3271: There was a mismatch between the processor architecture of the project being built "AMD64" and the processor architecture, "x86", of the implementation file "C:\Users\krsiler\git\fluentui-react-native\node_modules\react-native-windows\Microsoft.ReactNative\target\Microsoft.ReactNative\Microsoft.ReactNative.dll" for "C:\Users\krsiler\git\fluentui-react-native\node_modules\react-native-windows\Microsoft.ReactNative\target\Microsoft.ReactNative\Microsoft.ReactNative.winmd". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and implementation file, or choose a winmd file with an implementation file that has a processor architecture which matches the targeted processor architecture of your project. [C:\Users\krsiler\git\fluentui-react-native\node_modules\@react-native-community\slider\windows\SliderWindows\SliderWindows.vcxproj]. Check your build configuration.
 i Gathering MSBuild data for telemetry.
 i Loading properties from msbuildproperties.g.json
Command failed with error MSBuildError: 5:10>C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2302,5): error MSB3271: There was a mismatch between the processor architecture of the project being built "AMD64" and the processor architecture, "x86", of the implementation file "C:\Users\krsiler\git\fluentui-react-native\node_modules\react-native-windows\Microsoft.ReactNative\target\Microsoft.ReactNative\Microsoft.ReactNative.dll" for "C:\Users\krsiler\git\fluentui-react-native\node_modules\react-native-windows\Microsoft.ReactNative\target\Microsoft.ReactNative\Microsoft.ReactNative.winmd". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and implementation file, or choose a winmd file with an implementation file that has a processor architecture which matches the targeted processor architecture of your project. [C:\Users\krsiler\git\fluentui-react-native\node_modules\@react-native-community\slider\windows\SliderWindows\SliderWindows.vcxproj]
```

Any ideas on how to resolve it? I think we ran into something similar during the 0.66 update but I can't remember what we did to get passed it.

Saadnajmi commented:
> @rurikoaraki @Saadnajmi I'm getting pretty close to passing all checks but the Windows PR keeps failing. When I run `yarn windows --logging` locally, I get this error:
> 
> ```
>  × Building Solution
>  × Build failed with message 5:10>C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2302,5): error MSB3271: There was a mismatch between the processor architecture of the project being built "AMD64" and the processor architecture, "x86", of the implementation file "C:\Users\krsiler\git\fluentui-react-native\node_modules\react-native-windows\Microsoft.ReactNative\target\Microsoft.ReactNative\Microsoft.ReactNative.dll" for "C:\Users\krsiler\git\fluentui-react-native\node_modules\react-native-windows\Microsoft.ReactNative\target\Microsoft.ReactNative\Microsoft.ReactNative.winmd". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and implementation file, or choose a winmd file with an implementation file that has a processor architecture which matches the targeted processor architecture of your project. [C:\Users\krsiler\git\fluentui-react-native\node_modules\@react-native-community\slider\windows\SliderWindows\SliderWindows.vcxproj]. Check your build configuration.
>  i Gathering MSBuild data for telemetry.
>  i Loading properties from msbuildproperties.g.json
> Command failed with error MSBuildError: 5:10>C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2302,5): error MSB3271: There was a mismatch between the processor architecture of the project being built "AMD64" and the processor architecture, "x86", of the implementation file "C:\Users\krsiler\git\fluentui-react-native\node_modules\react-native-windows\Microsoft.ReactNative\target\Microsoft.ReactNative\Microsoft.ReactNative.dll" for "C:\Users\krsiler\git\fluentui-react-native\node_modules\react-native-windows\Microsoft.ReactNative\target\Microsoft.ReactNative\Microsoft.ReactNative.winmd". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and implementation file, or choose a winmd file with an implementation file that has a processor architecture which matches the targeted processor architecture of your project. [C:\Users\krsiler\git\fluentui-react-native\node_modules\@react-native-community\slider\windows\SliderWindows\SliderWindows.vcxproj]
> ```
> 
> Any ideas on how to resolve it? I think we ran into something similar during the 0.66 update but I can't remember what we did to get passed it.

I remember we had to pass the arch here, or else the command would fail. Should we not pass x64? https://github.com/microsoft/fluentui-react-native/blob/main/.ado/templates/e2e-testing-uwp.yml#L44

ksiler commented:
> > @rurikoaraki @Saadnajmi I'm getting pretty close to passing all checks but the Windows PR keeps failing. When I run `yarn windows --logging` locally, I get this error:
> > ```
> >  × Building Solution
> >  × Build failed with message 5:10>C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2302,5): error MSB3271: There was a mismatch between the processor architecture of the project being built "AMD64" and the processor architecture, "x86", of the implementation file "C:\Users\krsiler\git\fluentui-react-native\node_modules\react-native-windows\Microsoft.ReactNative\target\Microsoft.ReactNative\Microsoft.ReactNative.dll" for "C:\Users\krsiler\git\fluentui-react-native\node_modules\react-native-windows\Microsoft.ReactNative\target\Microsoft.ReactNative\Microsoft.ReactNative.winmd". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and implementation file, or choose a winmd file with an implementation file that has a processor architecture which matches the targeted processor architecture of your project. [C:\Users\krsiler\git\fluentui-react-native\node_modules\@react-native-community\slider\windows\SliderWindows\SliderWindows.vcxproj]. Check your build configuration.
> >  i Gathering MSBuild data for telemetry.
> >  i Loading properties from msbuildproperties.g.json
> > Command failed with error MSBuildError: 5:10>C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2302,5): error MSB3271: There was a mismatch between the processor architecture of the project being built "AMD64" and the processor architecture, "x86", of the implementation file "C:\Users\krsiler\git\fluentui-react-native\node_modules\react-native-windows\Microsoft.ReactNative\target\Microsoft.ReactNative\Microsoft.ReactNative.dll" for "C:\Users\krsiler\git\fluentui-react-native\node_modules\react-native-windows\Microsoft.ReactNative\target\Microsoft.ReactNative\Microsoft.ReactNative.winmd". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and implementation file, or choose a winmd file with an implementation file that has a processor architecture which matches the targeted processor architecture of your project. [C:\Users\krsiler\git\fluentui-react-native\node_modules\@react-native-community\slider\windows\SliderWindows\SliderWindows.vcxproj]
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Any ideas on how to resolve it? I think we ran into something similar during the 0.66 update but I can't remember what we did to get passed it.
> 
> I remember we had to pass the arch here, or else the command would fail. Should we not pass x64? https://github.com/microsoft/fluentui-react-native/blob/main/.ado/templates/e2e-testing-uwp.yml#L44

It doesn't make a difference for me locally. I still get the same error.

Saadnajmi commented:
> > > @rurikoaraki @Saadnajmi I'm getting pretty close to passing all checks but the Windows PR keeps failing. When I run `yarn windows --logging` locally, I get this error:
> > > ```
> > >  × Building Solution
> > >  × Build failed with message 5:10>C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2302,5): error MSB3271: There was a mismatch between the processor architecture of the project being built "AMD64" and the processor architecture, "x86", of the implementation file "C:\Users\krsiler\git\fluentui-react-native\node_modules\react-native-windows\Microsoft.ReactNative\target\Microsoft.ReactNative\Microsoft.ReactNative.dll" for "C:\Users\krsiler\git\fluentui-react-native\node_modules\react-native-windows\Microsoft.ReactNative\target\Microsoft.ReactNative\Microsoft.ReactNative.winmd". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and implementation file, or choose a winmd file with an implementation file that has a processor architecture which matches the targeted processor architecture of your project. [C:\Users\krsiler\git\fluentui-react-native\node_modules\@react-native-community\slider\windows\SliderWindows\SliderWindows.vcxproj]. Check your build configuration.
> > >  i Gathering MSBuild data for telemetry.
> > >  i Loading properties from msbuildproperties.g.json
> > > Command failed with error MSBuildError: 5:10>C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2302,5): error MSB3271: There was a mismatch between the processor architecture of the project being built "AMD64" and the processor architecture, "x86", of the implementation file "C:\Users\krsiler\git\fluentui-react-native\node_modules\react-native-windows\Microsoft.ReactNative\target\Microsoft.ReactNative\Microsoft.ReactNative.dll" for "C:\Users\krsiler\git\fluentui-react-native\node_modules\react-native-windows\Microsoft.ReactNative\target\Microsoft.ReactNative\Microsoft.ReactNative.winmd". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and implementation file, or choose a winmd file with an implementation file that has a processor architecture which matches the targeted processor architecture of your project. [C:\Users\krsiler\git\fluentui-react-native\node_modules\@react-native-community\slider\windows\SliderWindows\SliderWindows.vcxproj]
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Any ideas on how to resolve it? I think we ran into something similar during the 0.66 update but I can't remember what we did to get passed it.
> > 
> > 
> > I remember we had to pass the arch here, or else the command would fail. Should we not pass x64? https://github.com/microsoft/fluentui-react-native/blob/main/.ado/templates/e2e-testing-uwp.yml#L44
> 
> It doesn't make a difference for me locally. I still get the same error.

Another thing to try is passing `use-nuget` to the `install-windows-test-app` command in the ado yml file I linked above. It will use prebuilt nugets for the RNW binaries instead of building from source I believe. I've personally switched back-and forth having that flag in previous FURN RN updates based on whether the windows CI would build or not (😅). Beyond that, I'm not very well versed in the windows build system and how to resolve issues. Sometimes it's a react-native-test-app bug and we've resolved it by reaching out to them.

rurikoaraki commented:
> > > > @rurikoaraki @Saadnajmi I'm getting pretty close to passing all checks but the Windows PR keeps failing. When I run `yarn windows --logging` locally, I get this error:
> > > > ```
> > > >  × Building Solution
> > > >  × Build failed with message 5:10>C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2302,5): error MSB3271: There was a mismatch between the processor architecture of the project being built "AMD64" and the processor architecture, "x86", of the implementation file "C:\Users\krsiler\git\fluentui-react-native\node_modules\react-native-windows\Microsoft.ReactNative\target\Microsoft.ReactNative\Microsoft.ReactNative.dll" for "C:\Users\krsiler\git\fluentui-react-native\node_modules\react-native-windows\Microsoft.ReactNative\target\Microsoft.ReactNative\Microsoft.ReactNative.winmd". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and implementation file, or choose a winmd file with an implementation file that has a processor architecture which matches the targeted processor architecture of your project. [C:\Users\krsiler\git\fluentui-react-native\node_modules\@react-native-community\slider\windows\SliderWindows\SliderWindows.vcxproj]. Check your build configuration.
> > > >  i Gathering MSBuild data for telemetry.
> > > >  i Loading properties from msbuildproperties.g.json
> > > > Command failed with error MSBuildError: 5:10>C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2302,5): error MSB3271: There was a mismatch between the processor architecture of the project being built "AMD64" and the processor architecture, "x86", of the implementation file "C:\Users\krsiler\git\fluentui-react-native\node_modules\react-native-windows\Microsoft.ReactNative\target\Microsoft.ReactNative\Microsoft.ReactNative.dll" for "C:\Users\krsiler\git\fluentui-react-native\node_modules\react-native-windows\Microsoft.ReactNative\target\Microsoft.ReactNative\Microsoft.ReactNative.winmd". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and implementation file, or choose a winmd file with an implementation file that has a processor architecture which matches the targeted processor architecture of your project. [C:\Users\krsiler\git\fluentui-react-native\node_modules\@react-native-community\slider\windows\SliderWindows\SliderWindows.vcxproj]
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > Any ideas on how to resolve it? I think we ran into something similar during the 0.66 update but I can't remember what we did to get passed it.
> > > 
> > > 
> > > I remember we had to pass the arch here, or else the command would fail. Should we not pass x64? https://github.com/microsoft/fluentui-react-native/blob/main/.ado/templates/e2e-testing-uwp.yml#L44
> > 
> > 
> > It doesn't make a difference for me locally. I still get the same error.
> 
> Another thing to try is passing `use-nuget` to the `install-windows-test-app` command in the ado yml file I linked above. It will use prebuilt nugets for the RNW binaries instead of building from source I believe. I've personally switched back-and forth having that flag in previous FURN RN updates based on whether the windows CI would build or not (😅). Beyond that, I'm not very well versed in the windows build system and how to resolve issues. Sometimes it's a react-native-test-app bug and we've resolved it by reaching out to them. https://github.com/microsoft/react-native-test-app/blob/23199ac4499a8ba62b8c945e3f28d9b318fd7549/CONTRIBUTING.md#windows

Hm I get a different error

Build failed with message 2:6>E:\fluentui-react-native\apps\fluent-tester\windows\packages\Microsoft.Windows.CppWinRT.2.0.211028.7\build\native\Microsoft.Windows.CppWinRT.targets(741,9): error MSB3073: The command ""E:\fluentui-react-native\apps\fluent-tester\windows\packages\Microsoft.Windows.CppWinRT.2.0.211028.7\build\native\..\..\bin\"cppwinrt @"x64\Debug\ReactTestApp.vcxproj.cppwinrt_ref.rsp"" exited with code 1. [E:\fluentui-react-native\apps\fluent-tester\node_modules\.generated\windows\ReactTestApp\ReactTestApp.vcxproj]. Check your build configuration.

I've tried installing the test app cleanly using yarn install-windows-test-app with both -use-nuget and not. Also we already pass the arch as part of `yarn windows` so that shouldn't be the issue... I'd probably reach out to the test app group about this.

ksiler commented:
I was able to resolve the windows build error by downgrading react-native-svg to 12.3.0 (was previously at 12.4.4 with dependency specified at "^12.3.0").

Now, for some reason, we're getting an assert where rex-win32 calls abort() on the SVG test page. More specifically, it seems to be stemming from the [Remotely Retrieved SVG](https://github.com/microsoft/fluentui-react-native/blob/main/apps/fluent-tester/src/TestComponents/Svg/SvgTest.tsx#L121) test case. Will look into the cause.

ksiler commented:
Okay, I found the issue with SVGs. There was a recent change made to react-native-win32 to align the HTTP module specifications to react-native-windows code and I don't think the native HTTP module has been updated to reflect it: https://github.com/microsoft/react-native-windows/pull/10628

We can either wait for the native module to be updated and bump the rex-win32 version or downgrade @office-iss/react-native-win32 to version 0.68.5 for now and make a task to bump rex-win32 & react-native-win32 later when an update to the native module becomes available.

I'll downgrade react-native-win32 for now. @rurikoaraki let me know if you have any thoughts on this.

rurikoaraki commented:
@ksiler is it on someone's plate to fix it in native? downgrading seems fine for now as long as we don't need anything from the later versions

ksiler commented:
@rurikoaraki yes, @JunielKatarn should be looking into it.
